src/oatpp/network/Server.hpp

```diff
@@ -111,12 +114,21 @@ class Server : public base::Countable {
     return std::make_shared<Server>(connectionProvider, connectionHandler);
   }
 
+  /**
+   * Call &id:oatpp::network::ConnectionProvider::getConnection; in the loop and passes obtained Connection
+   * to &id:oatpp::network::ConnectionHandler;.
+   * @param conditional - Function that is called every mainloop iteration to check if the server should continue to run <br>
+   * Return true to let the server continue, false to shut it down.
+   */
+  void run(std::function<bool()> conditional = nullptr);
+
   /**
    * Call &id:oatpp::network::ConnectionProvider::getConnection; in the loop and passes obtained Connection
    * to &id:oatpp::network::ConnectionHandler;.
    * @param startAsNewThread - Start the server blocking (thread of callee) or non-blocking (own thread)
+   * @deprecated Deprecated since 1.3.0, will be removed in the next release.
    */
-  void run(bool startAsNewThread = false);
+  void run(bool startAsNewThread);
```

Comment:
I feel like we can drop it right away.
However, if you think it's better to keep it till the next release I'm ok with that too.

Reply:
Hm, this is a critical change in one of the core components. My guts tell me we should keep it and remove it in 1.3.0
However, I don't know if this feature is actually used that heavily. I want to ease the process of upgrading oat++ as much as possible. Therefore I think its our responsibility to warn users about breaking changes in minor releases (major releases are a whole other story). And since keeping this API alive until the next release does not hurt in any way (we don't have to implement compromises/adapters/workarounds) I assume this would be a nice move.